File: stockify_app_flutter/lib/feature/item/service/item_service.dart
import '../model/item.dart';
import '../model/item_filter_param.dart';

abstract class ItemService {
  void addItem(Item item);

  void updateItem(Item item);

  void deleteItem(int id);

  List<Item> getAllItems();

  List<Item> getFilteredItems(ItemFilterParams params);

  Item getItem(int id);
}

File: stockify_backend_golang/src/feature/item/service/item_service.go
package service

import "stockify_backend_golang/src/feature/item/model"

type ItemService interface {
	AddItem(item model.Item)
	GetAllItems() []model.Item
	GetItemById(id uint64) model.Item
	UpdateItem(item model.Item)
	DeleteItemById(id uint64)
}
